components/brave_today/browser/brave_news_controller.cc

```diff
@@ -313,6 +323,56 @@ void BraveNewsController::GetImageData(const GURL& padded_image_url,
           std::move(callback), is_padded));
 }
 
+void BraveNewsController::GetFavIconData(const std::string& publisher_id,
+                                         GetFavIconDataCallback callback) {
+  GetPublishers(base::BindOnce(
+      [](BraveNewsController* controller, const std::string& publisher_id,
+         GetFavIconDataCallback callback, Publishers publishers) {
+        // If the publisher doesn't exist, there's nothing we can do.
+        const auto& it = publishers.find(publisher_id);
+        if (it == publishers.end()) {
+          std::move(callback).Run(absl::nullopt);
+          return;
+        }
+
+        // If we have a FavIcon url, use that.
+        const auto& publisher = it->second;
+        if (publisher->favicon_url) {
+          controller->GetImageData(publisher->favicon_url.value(),
+                                   std::move(callback));
+          return;
+        }
+
+        auto source_url = publisher->site_url.is_valid()
+                              ? publisher->site_url
+                              : publisher->feed_source;
+        favicon_base::IconTypeSet icon_types{
+            favicon_base::IconType::kFavicon,
+            favicon_base::IconType::kTouchIcon};
+        controller->favicon_service_->GetRawFaviconForPageURL(
+            source_url, icon_types,
+
+            kDesiredFaviconSizePixels, true,
```

Comment:
How about passing `0` as a `desired_size_in_pixel`?
According to the comment, it'll give largest one with `0`.

Reply:
I think we probably don't want the largest one (it could be very large, as this include apple touch icons), and we're just displaying them as a 24x24px image, so I feel like @2x is quite a nice balance of not pixelated & not too big either.

In addition, the icon at size 256x256px isn't necessarily the same as a smaller favicon (it could have significantly more detail which might look odd when the icon is resized).

We also store the image in memory (as a blob) so a nice side effect of this is that the browser will resize any big images for us.

Reply:
Got it. Thanks for the kind explanation! :)